### PR TITLE
Cleanup for the trimming tree logic

### PR DESCRIPTION
**Pull request by GuyAv46**
**Describe the changes in the pull request**

Code cleanup and reduce code duplication

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: src/numeric_index.c [hunk 1]
```diff
@@ -458,10 +451,8 @@ int NumericRangeNode_RemoveChild(NumericRangeNode **node, NRN_AddRv *rv) {
   }
 
   // run recursively on both children
-  int rvRight = NumericRangeNode_RemoveChild(&n->right, rv);
-  int rvLeft = NumericRangeNode_RemoveChild(&n->left, rv);
-  NumericRangeNode *rightChild = n->right;
-  NumericRangeNode *leftChild = n->left;
+  bool rvRight = NumericRangeNode_RemoveChild(&n->right, rv);
```
Comment (kei-nan) [inline src/numeric_index.c:454]:
```suggestion
const  bool rvRight = NumericRangeNode_RemoveChild(&n->right, rv);
```

File: src/numeric_index.c [hunk 2]
```diff
@@ -458,10 +451,8 @@ int NumericRangeNode_RemoveChild(NumericRangeNode **node, NRN_AddRv *rv) {
...
+  bool rvLeft = NumericRangeNode_RemoveChild(&n->left, rv);
```
Comment (kei-nan) [inline src/numeric_index.c:455]:
```suggestion
  const bool rvLeft = NumericRangeNode_RemoveChild(&n->left, rv);
```

File: src/numeric_index.c [hunk 3]
```diff
@@ -471,38 +462,28 @@ int NumericRangeNode_RemoveChild(NumericRangeNode **node, NRN_AddRv *rv) {
...
     // left child is empty, save right as parent
-    rm_free(n);
-    *node = rightChild;
-    NumericRangeNode_Free(leftChild, rv);
+    *node = n->right;
+    n->right = NULL; // avoid freeing it
   }
-  return CHILD_NOT_EMPTY;
+  NumericRangeNode_Free(n, rv); // free the current node and its potential subtree
+  return (rvRight == CHILD_NOT_EMPTY || rvLeft == CHILD_NOT_EMPTY) ? CHILD_NOT_EMPTY : CHILD_EMPTY;
```
Comment (kei-nan) [inline src/numeric_index.c:486]: isn't the `rvRight` and `rvLeft` types is bool?
Maybe give them a better name and simplify this statement?

File: src/numeric_index.c [hunk 4]
```diff
@@ -471,38 +462,28 @@ int NumericRangeNode_RemoveChild(NumericRangeNode **node, NRN_AddRv *rv) {
...
-    // right child is empty, save left as parent
-    rm_free(n);
-    *node = leftChild;
-    NumericRangeNode_Free(rightChild, rv);
+    // right child is empty, save left as parent (might be empty)
+    *node = n->left;
+    n->left = NULL; // avoid freeing it
   } else {
     // left child is empty, save right as parent
-    rm_free(n);
```
Comment (kei-nan) [inline src/numeric_index.c:501]: where did the rm_free(n) get moved to?

Comment (kei-nan) [reply src/numeric_index.c:501]: nvm, found it

File: src/numeric_index.c (hunk 3 above)
Comment (GuyAv46) [reply src/numeric_index.c:486]: The compiler simplifies this line to `ret rvRight and rvLeft`. I think this line is more readable for humans and does not require explaining why we can do `rvRight && rvLeft`

File: src/numeric_index.c (hunk 4 above)
Comment (GuyAv46) [reply src/numeric_index.c:501]: this `rm_free` call, along with the previous call to `removeRange`, was moved to the last call to `NumericRangeNode_Free` which does exactly that

**Pull request merged**